### DISCUSSION TITLE
Background color changed after changing factory.setDrawingPropertiesForItem

### DISCUSSION THREAD

**Discussion by futurexdesign** [davetcc/tcMenu#329] [General]
I have an interface as such:

TitleBar
EditableTextMenuItem (displaying a timer countdown)
Grid of 4 action items represented by icons. 

I am using DarkModeModern on an LCD.
I want my timer section to be centered, with a white background and black text, but the rest of the ui to retain the black background.  The issue I'm having is, after I set the drawing properties for the timer menu entry, all of the "empty" space below the grid row has become white. 

I'm calling 
```
    MenuPadding perSidePadding(4, 4, 4, 4); 
    const color_t timmerPalette[] = {RGB(0, 0, 0), RGB(255,255,255), RGB(255,255,255), RGB(255,255,255)};


    factory.setDrawingPropertiesForItem(ItemDisplayProperties::COMPTYPE_ITEM, menuTimeRemaining.getId(), timmerPalette, perSidePadding, MenuFontDef(nullptr, 7).fontData , MenuFontDef(nullptr, 7).fontMag, 2, 60, GridPosition::JUSTIFY_CENTER_VALUE_ONLY, MenuBorder(1));
``` 
in setup after `setupMenu()`. I'm just not entirely sure why the remaining background has become white.  As a test, I tried adding another dummy menu time after the grid row, which renders correctly (in black background), but doesn't seem to fix the rest of the unfilled white space below the last menu. 

Screenshot: 
![screenshot_1007](https://user-images.githubusercontent.com/8275093/231331495-36dd6db9-52c9-446d-a712-4563d37bfd91.png)

Any suggestions on what to try would be appreciated.

**Comment by davetcc** [comment]
Just took a look at the code and I can see what the issue is, it is because the "fill" background drawing code is using your menu item as the background because of an oversight in the design. Up to now we've never seen the first menu item being overridden, and as it is also the "first" in the list, when we look for the background color that gets picked up instead of the default. This will be fixed in 3.2 that's not too far away. Converted to issue.

**Comment by davetcc** [comment]
See https://github.com/davetcc/tcMenu/issues/330 (tracking issue) and https://github.com/davetcc/tcMenuLib/issues/190 (actual issue in library repo)

**Reply by futurexdesign** [reply]
Thanks for the quick follow up.  I thought I was going crazy there for awhile.
